torch/fx/experimental/Partitioner.py

```diff
@@ -245,18 +259,47 @@ def dump_partition_DAG(self, module_with_submodules: GraphModule) -> DAG:
                 output_nodes = list(node.users)
             else:
                 output_nodes = [node]
-            dag.create_node(node, list(input_nodes), output_nodes)
+            dag.create_node(node, list(input_nodes), output_nodes, [])
+        # Assign logical devices to each node
+        # For a sparse nn partition,
+        # each embedding submodule has a different logical device.
+        # All the non-embedding submodules are assigned to all embedding submodule devices
+        if is_sparse_nn:
+            device_count = 0
+            embedding_devices = []
+            for i in range(len(dag.nodes)):
+                partition_id = int(dag.nodes[i].submodule_node.name.split('_')[-1])
```

Comment:
nit: you can use `rsplit` with a `maxsplit` to prevent splitting the entire string:
```
                partition_id = int(dag.nodes[i].submodule_node.name.rsplit('_', 1)[-1])
```